pom.xml

```diff
@@ -181,6 +184,18 @@
         <artifactId>snappy-java</artifactId>
         <version>${snappy.version}</version>
       </dependency>
+
+      <!-- Remove after Beam gets the latest gax version -->
+      <dependency>
+        <groupId>com.google.api</groupId>
+        <artifactId>gax-grpc</artifactId>
+        <version>${gax.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>com.google.api</groupId>
+        <artifactId>gax</artifactId>
+        <version>${gax.version}</version>
```

Comment:
Can you file an issue for this with details on why this was required?
Please also link the issue here with a TODO in a comment.

CC: @damccorm

Reply:
Filed a task in Beam: https://github.com/apache/beam/issues/30700 and added a TODO here.

Will resolve this TODO once that task is done and dataflow's beam is updated to 2.56 (which is expected to have this change).

There are no concerning details around this -- This PR updates the Spanner Java Client library to 6.61, which requires the latest version of gax. Without updating gax, we see errors like `class com.google.cloud.spanner.v1.stub.SpannerStubSettings overrides final method com.google.api.gax.rpc.StubSettings.getEndpoint()Ljava/lang/String;`. These errors are not related to the FLOAT32 type support -- it is just correcting the dependencies here.

Since dataflow's gax dependency seems to be coming from what Beam is using [1], we will not be able to update the spanner library version in dataflow unless the gax version is updated in Beam (which should automatically happen when the spanner library version is updated in Beam too).

(sorry if the details are wrong / confusing, I'm unfamiliar with development around maven and managing its dependencies; PLMK if you have a better suggestion here)

[1] or rather, the spanner component in Dataflow templates is using a Beam version that uses an older gax version, that's incompatible with spanner client lib 6.61. Ideally, we would not see this issue if Beam already uses the latest spanner client library (which I believe is how Dataflow templates' development is expected to be done -- Beam first and then Dataflow templates, but we're trying to skip that step here for a faster rollout)

Reply:
Thank you!